src/main/scala/com/codacy/api/client/CodacyClient.scala

```diff
@@ -92,7 +96,13 @@ class CodacyClient(
       parseJsonAs[T](body)
     } catch {
       case NonFatal(ex) =>
-        RequestResponse.failure(s"Error doing a post to ${url} : ${ex.getMessage}")
+        val zero: Option[Int] = Some(0)
+        (noRetries, zero) match {
```

Comment:
Why are you creating a value as `Some(0)` if it's a constant? can't you inline the value `0` at use site?

Reply:
I just did a new commit. What do you think?